components/leave_team_modal/index.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+
+import {emitLeaveTeam, toggleSideBarRightMenuAction} from 'actions/global_actions.jsx';
```

Comment:
Instead of using `emitLeaveTeam`, please use directly the redux action `import {removeUserFromTeam} from 'mattermost-redux/actions/teams';`

Doing so, you may delete that global actions' [emitLeaveTeam](https://github.com/mattermost/mattermost-webapp/blob/415bec8ce56dbc21f89615f7b5b6bd0397a2ce3d/actions/global_actions.jsx#L165).